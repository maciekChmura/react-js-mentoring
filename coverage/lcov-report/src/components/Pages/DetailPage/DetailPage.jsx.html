<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Pages/DetailPage/DetailPage.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/components/Pages/DetailPage</a> DetailPage.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import { fetchById } from '../../../utils/dataLoaders';
import MovieDetails from '../../Detail/MovieDetails';
import Detail from './DetailPage.Styles';
&nbsp;
class DetailPage extends Component {
  state = {
    movie: ''
  };
&nbsp;
  componentDidMount = () =&gt; {
    fetchById(12).then(<span class="fstat-no" title="function not covered" >da</span>ta =&gt; <span class="cstat-no" title="statement not covered" >this.setState(<span class="fstat-no" title="function not covered" >st</span>ate =&gt; (<span class="cstat-no" title="statement not covered" >{ movie: data })</span>))</span>;
  };
&nbsp;
  render() {
    const { movie } = this.state;
    return (
      &lt;Detail&gt;
        {movie ? <span class="branch-0 cbranch-no" title="branch not covered" >&lt;MovieDetails details={movie} /&gt; </span>: &lt;p&gt;loading&lt;/p&gt;}
      &lt;/Detail&gt;
    );
  }
}
&nbsp;
export default DetailPage;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Dec 21 2018 15:25:26 GMT+0100 (STD)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
