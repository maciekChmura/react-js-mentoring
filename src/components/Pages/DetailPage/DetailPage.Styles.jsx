import styled from 'styled-components';
import { backgroundImage } from '../../styleVariables';

const Detail = styled.div`
  width: 900px;
  margin: auto;
  padding: 20px;
  background-color: black;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-image: url(${backgroundImage});
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
`;

export default Detail;
