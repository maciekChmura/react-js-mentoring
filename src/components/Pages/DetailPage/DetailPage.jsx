import React, { Component } from 'react';
import { fetchById } from '../../../utils/dataLoaders';
import MovieDetails from '../../Detail/MovieDetails';
import Detail from './DetailPage.Styles';

class DetailPage extends Component {
  // state = {
  //   movie: ''
  // };

  // componentDidMount = () => {
  //   const { movieId } = this.props;
  //   fetchById(movieId).then(data => this.setState({ movie: data }));
  // };

  render() {
    // const { movie } = this.state;
    const { movieData } = this.props;
    return (
      <Detail>
        {movieData ? <MovieDetails details={movieData} /> : <p>loading</p>}
      </Detail>
    );
  }
}

export default DetailPage;
