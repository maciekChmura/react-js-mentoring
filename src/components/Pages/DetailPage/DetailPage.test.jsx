import React from 'react';
import renderer from 'react-test-renderer';
import DetailPage from './DetailPage';

test('DetailPage renders', () => {
  const component = renderer.create(<DetailPage />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
