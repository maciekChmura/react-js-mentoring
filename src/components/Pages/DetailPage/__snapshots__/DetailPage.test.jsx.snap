// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPage renders 1`] = `
<div
  className="DetailPageStyles__Detail-sc-4qdnn8-0 kinPBU"
>
  <p>
    loading
  </p>
</div>
`;
